dev-server/ui-playground/shared/components/PlaygroundApp.js

```diff
@@ -1,10 +1,17 @@
 import { SvgIcon } from '@hypothesis/frontend-shared';
 
-import MenuPatterns from './patterns/MenuPatterns';
 import SharedButtonPatterns from './patterns/SharedButtonPatterns';
 
 import { useRoute } from '../router';
 
+/**
+ * @typedef PlaygroundRoute - Route "handler" that provides a component (function)
+ *   that should be rendered for the indicated route
+ * @prop {RegExp|string} route
+ * @prop {string} title
+ * @prop {function} component
```

Comment:
Hokay. I spent some time looking at preact's types, and didn't find what I was looking for. I found `FunctionalComponent`, but that refers to an _instance_ of a functional component. Seems like there's _got_ to be a type for this...

Reply:
I think `FunctionComponent<{}>` is correct. The `{}` in angle-brackets is a type that specifies the props. In order to get the checking of `routes` below to work correctly I found it was necessary to change it to:

```js
/** @type {PlaygroundRoute[]} */
const routes = [
  {
    route: /^\/?$/,
    title: 'Home',
    component: HomeRoute,
  },
  {
    route: '/shared-buttons',
    title: 'Buttons',
    component: SharedButtonPatterns,
  },
];
```

It looks like casting the value before assigning to `routes` somehow ends up allowing invalid `component` values to pass checking. That's unexpected but I haven't dived into why that happens.